interfaces/builtin/media_hub.go

```diff
@@ -0,0 +1,301 @@
+    member=PropertiesChanged
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`)
+
+var mediaHubConnectedSlotAppArmorClassic = []byte(`
+# Allow unconfined clients to interact with the player on classic
+dbus (receive)
+    bus=session
+    path=/core/ubuntu/media/Service
+    peer=(label=unconfined),
+
+dbus (receive)
+    bus=session
+    interface=org.freedesktop.DBus.Introspectable
+    peer=(label=unconfined),
+`)
+
+var mediaHubConnectedPlugAppArmor = []byte(`
+# Description: Allow using media-hub service. Reserved because this gives
+#  privileged access to the service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+# Allow clients to manage Player sessions
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="{Create,Detach,Reattach,Destroy,CreateFixed,Resume}Session"
+	peer=(label=###SLOT_SECURITY_TAGS###),
+
+# Allow clients to pause all other sessions
+dbus (send)
+    bus=session
+    path=/core/ubuntu/media/Service
+    interface=core.ubuntu.media.Service
+    member="PauseOtherSessions"
+	peer=(label=###SLOT_SECURITY_TAGS###),
+
+# Allow clients to query service properties
+dbus (send)
+    bus=system
+    path=/core/ubuntu/media/Service
+    interface=org.freedesktop.DBus.Properties
+    member="{Get,Set}"
+    peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive)
+   bus=system
+   path=/core/ubuntu/media/Service
+   interface=org.freedesktop.DBus.Properties
+   member=PropertiesChanged
+   peer=(label=###SLOT_SECURITY_TAGS###),
+
+dbus (receive)
+    bus=system
+    path=/
+    interface=org.freedesktop.DBus.ObjectManager
+    peer=(label=unconfined),
+`)
+
+var mediaHubPermanentSlotSecComp = []byte(`
+getsockname
+recvmsg
+sendmsg
+sendto
+`)
+
+var mediaHubConnectedPlugSecComp = []byte(`
+getsockname
+recvmsg
+sendmsg
+sendto
+`)
+
+var mediaHubPermanentSlotDBus = []byte(`
+<policy user="root">
+    <allow own="core.ubuntu.media.Service"/>
+    <allow send_destination="core.ubuntu.media.Service"/>
+    <allow send_interface="core.ubuntu.media.Service"/>
+</policy>
+`)
+
+var mediaHubConnectedPlugDBus = []byte(`
+<policy context="default">
+    <deny own="core.ubuntu.media.Service"/>
+    <allow send_destination="core.ubuntu.media.Service"/>
+    <allow send_interface="core.ubuntu.media.Service"/>
+</policy>
+`)
+
+type MediaHubInterface struct{}
+
+func (iface *MediaHubInterface) Name() string {
+	return "media-hub"
+}
+
+func (iface *MediaHubInterface) PermanentPlugSnippet(plug *interfaces.Plug, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	return nil, nil
+}
+
+func (iface *MediaHubInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor:
+		old := []byte("###SLOT_SECURITY_TAGS###")
+		new := slotAppLabelExpr(slot)
+		snippet := bytes.Replace(mediaHubConnectedPlugAppArmor, old, new, -1)
```

Comment:
What are your use cases for this interface? All snaps? classic + media-hub snap? classic + media-hub deb? All of them?

If classic + media-hub deb is one of them, I suggest adjusted this section and make the value of `new` conditional on release.OnClassic such that when on classic `new` is `unconfined`, otherwise it is slotAppLabelExpr(slot). See the modem-manager interface for an example.